sid/script.js

```diff
@@ -0,0 +1,633 @@
+        }
+        for (const block of allBlocks) {
+            block.style.backgroundColor = '';
+            block.textContent = '';
+            block.style.border = 'grey solid 1px';
+            block.style.color = 'white';
+        }
+    }, 1000);
+    setTimeout(function () {
+        mainHeading.textContent = 'Wordle Time!';
+        mainHeading.style.color = 'white';
+        mainHeading.style.opacity = '1';
+        mainHeading.style.backgroundColor = 'none';
+        mainHeading.style.transitionDuration = '1s';
+        mainHeading.style.transitionTimingFunction = 'ease-in';
+        scoreSection.style.fontSize = '1.25em';
+    }, 2000);
+    document.body.style.backgroundImage = '';
+    moreChances.style.display = 'none';
+    checkFullSquares();
+    imageFadeOut(img);
+    imageFadeOut(img2);
+    blockNumber = 0;
+    gameComplete = false;
+    pressedEnter = false;
+    wordChecked = false;
+    found = false;
+    rowChanged = false;
+    nextStartingPoint = 0;
+    rowNumber = -1;
+    rowsAdded = false;
+    gameLost = false;
+    random = dictionary[Math.floor(Math.random() * dictionary.length)];
+    console.log(random);
+    const allDivs = document.querySelectorAll('div');
+    for (const row of allDivs) {
+        if (row.classList.contains('newRow')) {
+            row.remove();
+        }
+    }
+    allBlocks = document.querySelectorAll('.squares');
+    rows = document.querySelectorAll('.row');
+}
+
+function checkFullSquares() {
+    for (const block of allBlocks) {
+        if (block.textContent !== '') {
+            block.style.border = '#fcfcfc solid 4px';
+        }
+        else {
+            block.style.border = 'grey solid 1px';
+        }
+    }
+}
+
+function wordInDict(word) {
+    if (dictionary.includes(word)) {
+        return true;
+    }
+    else {
+        return false;
+    }
+}
+
+function wordFound(event) {
+    if (gameComplete === false) {
+        if (event.type === 'click') {
+            wordTypeOut(event.target.textContent);
+        }
+        if (event.type === 'keyup' && event.key.length === 1 && event.key.match(/[a-z]/i) || event.key === 'Enter' || event.key === 'Backspace') {
+            buttonPress(event.key);
+            wordTypeOut(event.key);
+        }
+    }
+}
+
+
+//inspiration from An Object Is A's Medium Article on making images pop in and pop out
+function imageFadeIn(image) {
+    image.style.transitionDuration = '0.2s';
+    image.style.transform = 'scale(1.5)';
+
+    setTimeout(function () {
+        image.style.transitionDuration = '0.07s';
+        image.style.transform = 'scale(1)';
+    }, 200);
+}
+
+function imageFadeOut(image) {
+    image.style.transitionDuration = '0.2s';
+    image.style.transform = 'scale(1.5)';
+
+    setTimeout(function () {
+        image.style.transitionDuration = '0.07s';
+        image.style.transform = 'scale(0)';
+        image.src = '';
+    }, 200);
+
+}
+
+function changeButtonColor(button, color) {
```

Comment:
There're complicated logic here that `changeButtonColor` doesn't quite describe the purpose behind its function, e.g. as another developer if I call `changeButtonColor(button, "green")` I'd expect button would turn green rather than turn white.